test/dependency-manager/external.test.ts

```diff
@@ -0,0 +1,282 @@
+    });
+
+    const env_config = {
+      services: {
+        'architect/frontend:v1': {
+          dependencies: {
+            'architect/backend': {
+              interfaces: {
+                api: {
+                  host: 'api.localhost',
+                  port: 80
+                }
+              }
+            }
+          }
+        }
+      }
+    };
+
+    mock_fs({
+      '/stack/src/frontend/architect.json': JSON.stringify(frontend_config),
+      '/stack/arc.env.json': JSON.stringify(env_config),
+    });
+
+    const manager = await LocalDependencyManager.createFromPath(axios.create(), '/stack/arc.env.json', undefined, true);
+    const graph = manager.graph;
+    expect(graph.nodes).length(2);
+    expect(graph.nodes[0]).instanceOf(ServiceNode);
+    expect(graph.nodes[0].ref).eq('architect/frontend:v1')
+    expect(graph.nodes[1]).instanceOf(ExternalNode);
+    expect(graph.nodes[1].ref).eq('architect/frontend:v1.architect/backend:v2')
+    expect(graph.nodes[1].interfaces.api.host).eq('api.localhost');
+    expect(graph.nodes[1].interfaces.api.port).eq(80);
+    expect(graph.edges).length(1);
+  });
+
+  it('external datastore', async () => {
+    const frontend_config = {
+      name: 'architect/frontend',
+      interfaces: {
+        app: 8080
+      },
+      dependencies: {
+        'architect/backend': 'v1'
+      }
+    };
+
+    const backend_config = {
+      name: 'architect/backend',
+      interfaces: {
+        api: 8080
+      },
+      datastores: {
+        primary: {
+          image: 'postgres:11',
+          port: 5432
+        }
+      }
+    };
+
+    moxios.stubRequest(`/accounts/architect/services/frontend/versions/v1`, {
+      status: 200,
+      response: { tag: 'v1', config: frontend_config, service: { url: 'architect/frontend:v1' } }
+    });
+
+    moxios.stubRequest(`/accounts/architect/services/backend/versions/v1`, {
+      status: 200,
+      response: { tag: 'v1', config: backend_config, service: { url: 'architect/backend:v1' } }
+    });
+
+    const env_config = {
+      services: {
+        'architect/frontend:v1': {
+          dependencies: {
+            'architect/backend': {
+              datastores: {
+                primary: {
+                  host: 'db.localhost',
+                  port: 80
+                }
+              }
+            }
+          }
+        }
+      }
+    };
+
+    mock_fs({
+      '/stack/src/frontend/architect.json': JSON.stringify(frontend_config),
+      '/stack/arc.env.json': JSON.stringify(env_config),
+    });
+
+    const manager = await LocalDependencyManager.createFromPath(axios.create(), '/stack/arc.env.json', undefined, true);
+    const graph = manager.graph;
+    expect(graph.nodes).length(3);
+    expect(graph.nodes[0]).instanceOf(ServiceNode);
+    expect(graph.nodes[1]).instanceOf(ServiceNode);
+    expect(graph.nodes[2]).instanceOf(ExternalNode);
+    expect(graph.nodes[2].interfaces._default.host).eq('db.localhost');
```

Comment:
Aren't we in the process of deprecating the "_default" interface? If so, we might want to change the tests.

Reply:
This node is an external datastore which to fit the mold has _default as its interface. We should add a ticket to kill datastores if we are happy with this once its merged and integrated into api/app.